src/graphql/entries.ts

```diff
@@ -109,51 +111,83 @@ function isAsset(entity: EntryProps | (Entity & CollectionItem)): boolean {
   return 'linkType' in entity.sys && entity.sys.linkType === ASSET_TYPENAME;
 }
 
-function updateReferenceAssetField({
+async function updateReferenceAssetField({
   referenceFromPreviewApp,
   updatedReference,
   entityReferenceMap,
   locale,
 }: UpdateReferenceFieldProps) {
-  const match = entityReferenceMap.get(updatedReference.sys.id) as AssetProps | undefined;
-
-  if (!match) {
-    // if we don't have the asset we send a message back to the entry editor
-    // and it will then send the asset back in the entity reference map
-    // where we can calculate the asset on the next update message.
-    sendMessageToEditor({
-      action: 'ENTITY_NOT_KNOWN',
-      referenceEntityId: updatedReference.sys.id,
-      referenceContentType: ASSET_TYPENAME,
-    });
-    return;
-  }
+  console.log('>> update asset field', { referenceFromPreviewApp, updatedReference });
+  const { match } = await getReference(entityReferenceMap, updatedReference.sys.id, true);
+
+  console.log('>> match', match);
 
   return updateAsset(
     { ...referenceFromPreviewApp, ...updatedReference, __typename: ASSET_TYPENAME },
-    match,
+    match as AssetProps,
     locale
   );
 }
 
-function updateReferenceEntryField(
+async function getReference(
```

Comment:
This is the relevant part, currently it leads to overfetching, we need to extract this into a singleton to load things